### ISSUE TITLE
CSV export is failing due to old dependency on phpoffice.

### ISSUE THREAD

**Issue by hbel** [humhub/humhub#4536]
Hi, we just stumbled over an error in the task csv export when using php 7.4. We run into a yii exception every time to export gets called: _Trying to access array offset on value of type float_ 

It seems like HumHub is using an old dependency to phpoffice/phpspreadsheet, since the error was fixed there last year in September: https://github.com/PHPOffice/PhpSpreadsheet/commit/a691516664f48790bec8e4f2005e37cc0215cd82#diff-76675576954dd2377f05111ac355f3cf5d2b41a1649a037612ba09d9a34d9339

I cannot decide whether it’s problematic to upgrade this dependency, I can just confirm the error gets fixed by the above change, since we tried that out on our installations manually, but I would really appreciate if someone from the humhub dev team could take a look at this.

| Q                | A
| ---------------- | ---
| HumHub version   | 1.6.4
| PHP version      | 7.4.3
| Operating system | Linux

```
yii\base\ErrorException: Trying to access array offset on value of type float in /var/www/html/humhub/protected/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/DefaultValueBinder.php:58 Stack trace: #0 /var/www/html/humhub/protected/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/DefaultValueBinder.php(58): yii\base\ErrorHandler->handleError() #1 /var/www/html/humhub/protected/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/DefaultValueBinder.php(36): PhpOffice\PhpSpreadsheet\Cell\DefaultValueBinder::dataTypeForValue() #2 /var/www/html/humhub/protected/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/Cell.php(184): PhpOffice\PhpSpreadsheet\Cell\DefaultValueBinder->bindValue() #3 /var/www/html/humhub/protected/humhub/components/export/SpreadsheetExport.php(336): PhpOffice\PhpSpreadsheet\Cell\Cell->setValue() #4 /var/www/html/humhub/protected/humhub/components/export/SpreadsheetExport.php(262): humhub\components\export\SpreadsheetExport->composeBodyRow() #5 /var/www/html/humhub/protected/modules/tasks/controllers/AbstractTaskController.php(68): humhub\components\export\SpreadsheetExport->export() #6 [internal function]: humhub\modules\tasks\controllers\AbstractTaskController->actionExport() #7 /var/www/html/humhub/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array() #8 /var/www/html/humhub/protected/vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams() #9 /var/www/html/humhub/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\base\Controller->runAction() #10 /var/www/html/humhub/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\base\Module->runAction() #11 /var/www/html/humhub/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\web\Application->handleRequest() #12 /var/www/html/humhub/index.php(25): yii\base\Application->run() #13 {main}
```

**Comment by ArchBlood** [comment]
Doesn't seem to be a problem in v1.7.0-beta.1 as I'm able to generate and download the CSV file of both user & tasks while using PHP 7.4.11 :thinking:

**Comment by hbel** [comment]
Maybe the dependencies are already updated in the beta? 1.6.4 is definitely using an older version of PhpSpreadsheet which contains error mentioned above.

**Comment by yurabakhtin** [comment]
@luke- I couldn't reproduce the bug:
<img width="1027" alt="test_csv_export" src="https://user-images.githubusercontent.com/6995524/96484744-1340f680-1245-11eb-8061-fe055fd8d7e1.png">

In `composer.json` we have `"phpoffice/phpspreadsheet": "^1.0",`
In the file `protected/vendor/phpoffice/phpspreadsheet/CHANGELOG.md` I see: `## 1.15.0 - 2020-10-11`

**Comment by ArchBlood** [comment]
> @luke- I couldn't reproduce the bug:
> <img width="1027" alt="test_csv_export" src="https://user-images.githubusercontent.com/6995524/96484744-1340f680-1245-11eb-8061-fe055fd8d7e1.png">
> 
> In `composer.json` we have `"phpoffice/phpspreadsheet": "^1.0",`
> In the file `protected/vendor/phpoffice/phpspreadsheet/CHANGELOG.md` I see: `## 1.15.0 - 2020-10-11`

Might be an environment issue? :thinking:

**Comment by luke-** [comment]
@hbel Any idea?

**Comment by hbel** [comment]
@luke We updated the installation from an older version. Maybe the dependencies were not correctly updated by the update process? If I take a look at the `protected/vendor` directory on our server, I see that the files were touched, but the files do not correspond to the correct version of phpoffice/Phpspreadsheet.

**Comment by luke-** [comment]
@hbel How did you update (updater, git oder package)? From which version? Normally our packages contain a complete `protected/vendor` pack.

**Comment by tinkat69** [comment]
On behalf of @hbel, the updates are done manually 1.6.3 -> 1.6.4 (package). But the problem was there with HH 1.6.3 and no difference with 1.6.4. On another installation in a different environment it’s the same: the export by csv or xlsx of tasks or users is no problem with HH 1.6.3 and PHP 7.2.(29) but when the PHP is updated to 7.4.(5), the same export errors as the other system occur (see images). On returning to PHP 7.2.29, there are no problems.
<img width="879" alt="export-issues_01" src="https://user-images.githubusercontent.com/69628628/96584221-379ce180-12de-11eb-9249-7d4b77a1dc29.png">
<img width="1233" alt="export-issues_02" src="https://user-images.githubusercontent.com/69628628/96584305-526f5600-12de-11eb-847f-a6221e42edb2.png">
<img width="1155" alt="export-issues_03" src="https://user-images.githubusercontent.com/69628628/96584326-58fdcd80-12de-11eb-9a18-7dcd536cce6f.png">
<img width="821" alt="export-issues_04" src="https://user-images.githubusercontent.com/69628628/96584342-5ef3ae80-12de-11eb-8f6b-176d27b484c8.png">
<img width="823" alt="export-issues_05" src="https://user-images.githubusercontent.com/69628628/96584382-6f0b8e00-12de-11eb-89e7-9102616a54a0.png">
<img width="1145" alt="export-issues_06" src="https://user-images.githubusercontent.com/69628628/96584401-75016f00-12de-11eb-92b5-6c9d18e765aa.png">

**Comment by luke-** [comment]
Does this problem still persist? Can you please also try it with a recent HumHub 1.7.2?

**Comment by tinkat69** [comment]
No. With one installation, we had the problem with 1.6.x when PHP was updated to 7.4.x. This was solved by adding/updating PHPOffice and PHPXML extensions. Since then, this installation (currently 1.7.2) has had no such csv or xlsx export issues.
Another installation while running 1.6.x had no csv or xlsx export issues when using PHP 7.2.29 but threw the errors when PHP was updated to 7.4.x.  This installation, now on HH1.7.2, has no csv or xlsx export issues with PHP 7.2.29 and I have just tested with PHP 7.4.5 - which previously caused errors - and the CSV/XLSX exports are working fine.
